### ISSUE TITLE
[browser] Allow dynamic `process.env` in browser mode

### ISSUE THREAD

**Issue by JCQuintas** [vitest-dev/vitest#6667]
### Clear and concise description of the problem

Currently `process.env.NODE_ENV` gets replaced by its actual value when `vitest` is run in **browser** mode.

This behaviour defeats some parts of component testing, when we want to actually test the behaviour of our component in different environments. Eg: by throwing in development vs warning in production.

### Suggested solution

Allow `process.env` to be dynamic in `browser mode`

### Additional context

If you need a quick testing example 🙃 
https://github.com/JCQuintas/vitest-browser-process-env

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

**Comment by sheremet-va** [comment]
`process.env` doesn't exist in the browser, so I don't think it is reasonable to polyfill it since it will introduce the difference between the production and development environments.

We could play around with it by replacing it with something like `__global_process_env__` and override it in `vi.stubEnv` or `import.meta.env.NODE_ENV`

**Comment by JCQuintas** [comment]
`process.env` already works in vitest browser mode. The issue seems to be that the builder/bundler parses `process.env.NODE_ENV` of the components to `"production"` when it runs.

If we destructure the variable and use it instead, then the tests in the example start working.

```tsx
  const { NODE_ENV } = process.env;
  if (NODE_ENV === 'production') {
    throw new Error('This is a production error');
  }
```

**Comment by sheremet-va** [comment]
> `process.env` already works in vitest browser mode.

Yeah, and it's a bug: https://github.com/vitest-dev/vitest/issues/6203

> The issue seems to be that the builder/bundler parses `process.env.NODE_ENV` of the components to `"production"` when it runs

This is just how Vite works; we can bypass it by transforming it to a different variable first.

**Comment by XiaMingYu77** [comment]
I also encountered issues while configuring `import.meta.env`in browser mode. The `vi.stubEnv` is not functional, and directly inputting values does not take effect in the main code.
